flink-core/src/test/java/org/apache/flink/testutils/ClassLoaderUtils.java

```diff
@@ -158,7 +170,7 @@ public ClassLoaderBuilder withParentClassLoader(ClassLoader classLoader) {
             return this;
         }
 
-        public URLClassLoader build() throws IOException {
+        public ClassLoaderBuilder compile() throws IOException {
```

Comment:
should we name it `writeAndCompile`?
Otherwise the naming is IMHO a bit confusing : it is named `compile` and then besides compilations also makes some writing based on other methods names